### ISSUE TITLE
Task doesn't fail after read.timeout.ms is exceeded

### ISSUE THREAD

**Issue by jpmarques66** [confluentinc/kafka-connect-elasticsearch#534]
TL;DR
Elasticsearch becomes unavailable in the middle of a connection, which makes read.timeout.ms be exceeded. I expected the task to fail, but the task keeps the status `RUNNING`.

Description:

Hello! I'm trying to set up this elasticsearch connector to interact with tenant-owned systems, hence I need it to be resilient to tenants' systems failure (as much as possible). 

While testing for elasticsearch unavailability, I tried killing **its only master node** and I get this error:
```
[2021-05-25 16:34:00,818] WARN Bulk request 58 failed. Retrying request. (io.confluent.connect.elasticsearch.ElasticsearchClient)
java.net.SocketTimeoutException: 3,000 milliseconds timeout on connection http-outgoing-22 [ACTIVE]
	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.timeout(HttpAsyncRequestExecutor.java:387)
	at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:92)
	at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:39)
	at org.apache.http.impl.nio.reactor.AbstractIODispatch.timeout(AbstractIODispatch.java:175)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.sessionTimedOut(BaseIOReactor.java:263)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.timeoutCheck(AbstractIOReactor.java:492)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.validate(BaseIOReactor.java:213)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:280)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591)
	at java.base/java.lang.Thread.run(Thread.java:834)
[2021-05-25 16:39:05,899] INFO [Consumer clientId=connector-consumer-simple-elasticsearch-connector-0, groupId=connect-simple-elasticsearch-connector] Member connector-consumer-simple-elasticsearch-connector-0-a0b0c673-c1d6-4ec8-b9c6-138bec30adcc sending LeaveGroup request to coordinator kafka-0.kafka-headless.kafka.svc.cluster.local:9092 (id: 2147483647 rack: null) due to consumer poll timeout has expired. This means the time between subsequent calls to poll() was longer than the configured max.poll.interval.ms, which typically implies that the poll loop is spending too much time processing messages. You can address this either by increasing max.poll.interval.ms or by reducing the maximum size of batches returned in poll() with max.poll.records. (org.apache.kafka.clients.consumer.internals.AbstractCoordinator)
```
The error seems ok, but the task doesn't fail as I expected

```
❯ curl -X GET localhost:8083/connectors/simple-elasticsearch-connector/status | jq
{
  "name": "simple-elasticsearch-connector",
  "connector": {
    "state": "RUNNING",
    "worker_id": "10.244.1.103:8083"
  },
  "tasks": [
    {
      "id": 0,
      "state": "RUNNING",
      "worker_id": "10.244.1.103:8083"
    }
  ],
  "type": "sink"
}
```

Is this supposed to happen? Thanks in advance!!! <3

**Comment by thomas-tomlinson** [comment]
I'm experiencing the same condition as well.  I was able to reproduce this with the TRACE level logging and it shows an interesting behavior in the retry portion.  I've got the max.retries set to 19 and didn't set the retry.backoff.ms, so it's default to 100.  I have a reverse load balancer on the kafka connect hosts that connects to Elasticsearch, so after indexing a few documents I just shut that off on the node hosting the single task I have configured.

This is what the log shows
```
[2021-07-15 07:26:52,555] DEBUG Putting 1 records to Elasticsearch. (io.confluent.connect.elasticsearch.ElasticsearchSinkTask)
[2021-07-15 07:26:52,555] TRACE Writing record from topic=trans_2021_07 partition=0 offset=496188 to Elasticsearch. (io.confluent.connect.elasticsearch.ElasticsearchSinkTask)
[2021-07-15 07:26:52,556] TRACE Adding record from topic=trans_2021_07 partition=0 offset=496188 to bulk processor. (io.confluent.connect.elasticsearch.ElasticsearchSinkTask)
[2021-07-15 07:26:52,564] WARN Bulk request 4 failed. Retrying request. (io.confluent.connect.elasticsearch.ElasticsearchClient)
[2021-07-15 07:26:52,575] TRACE Try retrying bulk request (attempt 1 of 19) (io.confluent.connect.elasticsearch.RetryUtil)
[2021-07-15 07:26:53,638] DEBUG Putting 0 records to Elasticsearch. (io.confluent.connect.elasticsearch.ElasticsearchSinkTask)
[2021-07-15 07:26:53,639] DEBUG Flushing data to Elasticsearch with the following offsets: {ES_SINK_TRANS-0=OffsetAndMetadata{offset=496189, leaderEpoch=null, metadata=''}} (io.confluent.connect.elasticsearch.ElasticsearchSinkTask)
```
So it looks like the retry process is entered, but then the behavior doesn't make a lot of sense.  There's only a single retry and then 0 records are put into Elasticsearch, then nothing.  There's actually more records waiting but nothing happens.

**Comment by yeikel** [comment]
Did this ever get resolved? I am also seeing this one along with #739 and I am using the latest version of the connector
